base_layer/core/src/base_node/state_machine_service/states/horizon_state_sync.rs

```diff
@@ -57,26 +57,46 @@ impl HorizonStateSync {
             Err(err) => return err.into(),
         };
 
-        let last_header = match shared.db.fetch_last_header().await {
-            Ok(h) => h,
-            Err(err) => return err.into(),
+        let sync_peers = &mut self.sync_peers;
+        // // Order sync peers according to accumulated difficulty
+        sync_peers.sort_by(|a, b| {
+            b.claimed_chain_metadata()
+                .accumulated_difficulty()
+                .cmp(&a.claimed_chain_metadata().accumulated_difficulty())
+        });
+        let remote_metadata = match sync_peers.first() {
+            Some(peer) => peer.claimed_chain_metadata(),
+            None => return StateEvent::FatalError("No sync peers".into()),
         };
 
-        let horizon_sync_height = local_metadata.horizon_block_height(last_header.height);
+        let horizon_sync_height =
+            local_metadata.pruned_height_at_given_chain_tip(remote_metadata.height_of_longest_chain());
+
+        // Determine if we need to sync horizon state
         if local_metadata.pruned_height() >= horizon_sync_height {
```

Comment:
this if function used to check if you need to go to block sync, or stay in in horizon sync based on what the sync_target_height is vs what you local pruned height is. . 

It now checks if the pruned height is more than the remote peer height - local pruned height.